sdk/src/sharedTest/java/io/customer/sdk/queue/QueueStorageIntegrationTest.kt

```diff
@@ -132,9 +134,74 @@ class QueueStorageIntegrationTest : BaseTest() {
         queueStorage.create(String.random, String.random, null, null)
         queueStorage.create(String.random, String.random, null, listOf(givenStartOfAnotherGroup))
 
+        val newlyCreatedInventoryItem3 = queueStorage.getInventory()[2]
+        val newlyCreatedInventoryItem4 = queueStorage.getInventory()[3]
+
         val itemsDeleted = queueStorage.deleteGroup(givenStartOfTheGroup.toString())
         itemsDeleted.count() shouldBeEqualTo 2
-        queueStorage.getInventory().count() shouldBeEqualTo 2
+        val inventory = queueStorage.getInventory()
+        inventory.count() shouldBeEqualTo 2
+        inventory.map { it.taskPersistedId } shouldBeEqualTo listOf(
+            newlyCreatedInventoryItem3.taskPersistedId,
+            newlyCreatedInventoryItem4.taskPersistedId
+        )
+    }
+
+    @Test
+    fun givenDeleteGroupTask_givenMembersTasksBelongToDifferentGroups_expectAllStartTasksAndTheirMembersToBeDeleted() {
+        val givenStartOfTheGroup = QueueTaskGroup.IdentifyProfile(String.random)
+        val givenStartOfAnotherGroup = QueueTaskGroup.RegisterPushToken(String.random)
+
+        queueStorage.create(String.random, String.random, givenStartOfTheGroup, null)
+        queueStorage.create(String.random, String.random, null, listOf(givenStartOfTheGroup))
+        queueStorage.create(
+            String.random,
+            String.random,
+            givenStartOfAnotherGroup,
+            listOf(givenStartOfTheGroup)
+        )
+        queueStorage.create(String.random, String.random, null, listOf(givenStartOfAnotherGroup))
+
+        val itemsDeleted = queueStorage.deleteGroup(givenStartOfTheGroup.toString())
+        itemsDeleted.count() shouldBeEqualTo 4
+        queueStorage.getInventory().count() shouldBeEqualTo 0
+    }
+
+    @Test
+    fun givenDeleteGroupTask_givenMembersTasksBelongToDifferentGroups_givenDifferentOrder_expectAllStartTasksAndTheirMembersToBeDeleted() {
+        val givenStartOfTheGroup = QueueTaskGroup.IdentifyProfile(String.random)
+        val givenStartOfAnotherGroup = QueueTaskGroup.RegisterPushToken(String.random)
+
+        queueStorage.create(String.random, String.random, null, listOf(givenStartOfTheGroup))
+        queueStorage.create(String.random, String.random, givenStartOfTheGroup, null)
```

Comment:
This scenario *should* never occur where a BQ task that is the start of a group exists *after* a BQ task that belongs to that group. 

So, I don't see this test function adding value. Do you see a reason to keep this test function?

Reply:
It's still an edge case that I like to keep, to make sure we safe guard against it.